### PR TITLE
Troubleshooting guides cleanup

### PR DESCRIPTION
**Pull request by colunira**
**Description**

Changes proposed in this pull request:

- Add more links about documented topis
- Add more tips about sidecar injections issues
- Simplify some commands

### CODE REVIEW

File: docs/service-mesh/10-03-connection-refused.md [hunk 1]
```diff
@@ -1,11 +1,11 @@
...
 type: Troubleshooting
 ---
 
-Mutual TLS (mTLS) is enabled in the Service Mesh by default. As a result, every element of the Service Mesh must have an Istio sidecar to allow TLS communication. If you don't want to use a sidecar, you can whitelist a service and disable TLS traffic for it by creating a [DestinationRule](https://istio.io/docs/reference/config/networking/destination-rule/).  
+Mutual TLS (mTLS) is enabled in the Service Mesh by default. As a result, every element of the Service Mesh must have an Istio sidecar with a valid TLS certificate to allow communication. Connection between a service with a sidecar and a service without a sidecar results in a `Connection reset by peer` or a `GOAWAY` response. 
 
 - To enable sidecar injection for Pods of existing Services, restart them after upgrading to Kyma 1.0 or higher.
-- To create a DestinationRule for a Service, follow the official [Istio documentation](https://istio.io/docs/reference/config/networking/destination-rule/).
+- If you don't want to use a sidecar, you can whitelist a Service and disable TLS traffic for it by creating a [DestinationRule](https://istio.io/docs/reference/config/networking/destination-rule/).
```
Comment (piotrmsc) [inline docs/service-mesh/10-03-connection-refused.md:9]: To tell all the clients(other sidecars) to do not use mTLS when connecting to my service we use DestinationRule. To accept PERMISSIVE connections for my service that has a sidecar we use Policy - at least for now as there are now in 1.5 new CRs for that, but this is out of scope as we don't have it yet. To disable sidecar injection we follow step bellow :)

File: docs/api-gateway-v2/10-02-connect-api-rule.md [hunk 2]
```diff
@@ -5,7 +5,7 @@ type: Troubleshooting
 
 ##  Basic troubleshooting
 
-API Gateway is a controller. It adds a status to the rules it processes. For basic troubleshooting, check the APIRule status:
+API Gateway is a controller. It adds a status to the rules it processes. For basic troubleshooting, check the APIRule [status](#custom-resource-api-rule-status-codes):
 
    ```bash
    kubectl describe apirules.gateway.kyma-project.io -n {NAMESPACE} {NAME}
```
Comment (tomekpapiernik) [inline docs/api-gateway-v2/10-02-connect-api-rule.md:11]: I'd rename this entire section. It's more about diagnosing what's wrong, not so much about troubleshooting anything aside from the "usual suspects" (ie. check if this or that is running).

File: docs/service-mesh/10-02-no-access.md [hunk 3]
```diff
@@ -36,8 +34,13 @@ If this solution doesn't work, you need to change the image of the Istio Ingress
     kubectl exec -ti -n istio-system $(kubectl get pod -l app=istio-ingressgateway -n istio-system -o name) -c istio-proxy -- netstat -lptnu
     ```
 
-4. If ports `80` and `443` are not used, restart the Pods of the Istio Ingress Gateway to force them to recreate their configuration:
+4. If ports `80` and `443` are not used, check logs of the Istio Ingress Gateway container `ingress-sds` for errors related to the certificates:
```
Comment (tomekpapiernik) [inline docs/service-mesh/10-02-no-access.md:37]:
```suggestion
4. If ports `80` and `443` are not used, check the logs of the Istio Ingress Gateway container for errors related to certificates. Run:
```

Comment (tomekpapiernik) [reply docs/service-mesh/10-02-no-access.md:37]: You don't really have to give the container name in the description - we have it in the command ;)

File: docs/service-mesh/10-02-no-access.md [hunk 4]
```diff
@@ -36,8 +34,13 @@ If this solution doesn't work, you need to change the image of the Istio Ingress
...
 
     ```bash
-    kubectl delete pod -l app=istio-ingressgateway -n istio-system
+    kubectl logs -n istio-system -l app=istio-ingressgateway -c ingress-sds
     ```
+   
+5. If you notice any issues regarding certificates, make sure `kyma-gateway-certs` and `kyma-gateway-certs-cacert` Secrets are available in the `istio-system` Namespace and they contain proper data. To regenerate a corrupted certificate, follow [this guide](components/security/#tutorials-update-tls-certificate). If you run Kyma on Gardener, follow [this guide](components/security/#troubleshooting-issues-with-certificates-on-gardener) instead.
```
Comment (tomekpapiernik) [inline docs/service-mesh/10-02-no-access.md:43]:
```suggestion
5. In case of certificate-related issues, make sure `kyma-gateway-certs` and `kyma-gateway-certs-cacert` Secrets are available in the `istio-system` Namespace and that they contain proper data. To regenerate a corrupted certificate, follow [this tutorial](components/security/#tutorials-update-tls-certificate). If you are running Kyma provisioned through Gardener, follow [this tutorial](components/security/#troubleshooting-issues-with-certificates-on-gardener).
```

Maybe add a command that shows how to check these secrets and see "what's in their bellies"?

File: docs/service-mesh/10-03-connection-refused.md [hunk 5]
```diff
@@ -1,11 +1,12 @@
 ---
-title: Connection refused errors after upgrade
+title: Connection refused errors
 type: Troubleshooting
 ---
 
-Mutual TLS (mTLS) is enabled in the Service Mesh by default. As a result, every element of the Service Mesh must have an Istio sidecar to allow TLS communication. If you don't want to use a sidecar, you can whitelist a service and disable TLS traffic for it by creating a [DestinationRule](https://istio.io/docs/reference/config/networking/destination-rule/).  
+Mutual TLS (mTLS) is enabled in the Service Mesh by default. As a result, every element of the Service Mesh must have an Istio sidecar with a valid TLS certificate to allow communication. Connection between a service with a sidecar and a service without a sidecar results in a `Connection reset by peer` or a `GOAWAY` response.
```
Comment (tomekpapiernik) [inline docs/service-mesh/10-03-connection-refused.md:6]:
```suggestion
Mutual TLS (mTLS) is enabled in the Service Mesh by default. As a result, every element of the Service Mesh must have an Istio sidecar with a valid TLS certificate to allow communication. Attempts to establish connection between a service with a sidecar and a service without a sidecar results in a `Connection reset by peer` or a `GOAWAY` response. 
```

File: docs/service-mesh/10-03-connection-refused.md [hunk 6]
```diff
@@ -1,11 +1,12 @@
...
 ---
 
-Mutual TLS (mTLS) is enabled in the Service Mesh by default. As a result, every element of the Service Mesh must have an Istio sidecar to allow TLS communication. If you don't want to use a sidecar, you can whitelist a service and disable TLS traffic for it by creating a [DestinationRule](https://istio.io/docs/reference/config/networking/destination-rule/).  
+Mutual TLS (mTLS) is enabled in the Service Mesh by default. As a result, every element of the Service Mesh must have an Istio sidecar with a valid TLS certificate to allow communication. Connection between a service with a sidecar and a service without a sidecar results in a `Connection reset by peer` or a `GOAWAY` response. 
 
 - To enable sidecar injection for Pods of existing Services, restart them after upgrading to Kyma 1.0 or higher.
-- To create a DestinationRule for a Service, follow the official [Istio documentation](https://istio.io/docs/reference/config/networking/destination-rule/).
+- To allow connections to a Service with a sidecar from Services without a sidecar, create a [Policy](https://istio.io/docs/reference/config/security/istio.authentication.v1alpha1/) in a `PERMISSIVE` mode.
```
Comment (tomekpapiernik) [inline docs/service-mesh/10-03-connection-refused.md:9]:
```suggestion
- To allow connections between a Service with a sidecar and Services without a sidecar, create an [Authenitcation Policy](https://istio.io/docs/reference/config/security/istio.authentication.v1alpha1/) in the `PERMISSIVE` mode.
```

File: docs/service-mesh/10-05-sidecar.md [hunk 7]
```diff
@@ -4,4 +4,37 @@ type: Troubleshooting
 ---
 
 
-Kyma has sidecar injection enabled by default - it injects sidecars to every Deployment on a cluster, without the need of labeling a Deployment or Namespace. For more information, read [this document](#details-sidecar-proxy-injection) or follow the [Istio documentation](https://istio.io/docs/ops/common-problems/injection/).
+Kyma has sidecar injection enabled by default - it injects sidecars to every Deployment on a cluster, without the need of labeling a Deployment or Namespace. For more information, read [this document](#details-sidecar-proxy-injection).
```
Comment (tomekpapiernik) [inline docs/service-mesh/10-05-sidecar.md:7]:
```suggestion
Kyma has sidecar injection enabled by default - a sidecar is injected to every Deployment in a cluster, without the need adding any labels. For more information, read [this document](#details-sidecar-proxy-injection).
```

File: docs/service-mesh/10-05-sidecar.md [hunk 8]
```diff
@@ -4,4 +4,37 @@ type: Troubleshooting
...
+If your Pod doesn't have a sidecar, although it should, follow this steps:
```
Comment (tomekpapiernik) [inline docs/service-mesh/10-05-sidecar.md:8]:
```suggestion
If a Pod doesn't have a sidecar and you did not disable sidecar injection on purpose, follow these steps to troubleshoot:
```

File: docs/service-mesh/10-05-sidecar.md [hunk 9]
```diff
@@ -4,4 +4,37 @@ type: Troubleshooting
...
+
+1. Check if sidecar injection is disabled in the Namespace of the Pod:
```
Comment (tomekpapiernik) [inline docs/service-mesh/10-05-sidecar.md:10]:
```suggestion
1. Check if sidecar injection is disabled in the Namespace of the Pod. Run this command to check the `istio-injection` label:
```

File: docs/service-mesh/10-05-sidecar.md [hunk 10]
```diff
@@ -4,4 +4,37 @@ type: Troubleshooting
...
+
+    ```bash
+    kubectl get namespaces {NAMESPACE} -o jsonpath='{ .metadata.labels.istio-injection }'
+    ```
+
+    Sidecar injection is disabled in this namespace if the output is `disabled`. Move Deployment to another Namespace or delete the label and restart the Pod.
```
Comment (tomekpapiernik) [inline docs/service-mesh/10-05-sidecar.md:16]:
```suggestion
    If the command returns `disabled` the sidecar injection is disabled in this Namespace. To add a sidecar to the Pod, move the Pod's deployment to a Namespace that has sidecar injection enabled, or remove the label from the Namespace and restart the Pod.
```

File: docs/service-mesh/10-05-sidecar.md [hunk 11]
```diff
@@ -4,4 +4,37 @@ type: Troubleshooting
...
+    
+    >**WARNING:** Removing the label from Namespace will result with injecting sidecars to all Pods inside of the Namespace.
```
Comment (tomekpapiernik) [inline docs/service-mesh/10-05-sidecar.md:18]:
```suggestion
    >**WARNING:** Removing the `istio-injection=disabled` label from Namespace results in injecting sidecars to all Pods inside of the Namespace.
```
... but only after the Pods are restarted? If so, pls add that.

Comment (tomekpapiernik) [reply docs/service-mesh/10-05-sidecar.md:18]: I might've gotten the label name wrong, please adjust if that's true.

File: docs/service-mesh/10-05-sidecar.md [hunk 12]
```diff
@@ -4,4 +4,37 @@ type: Troubleshooting
...
+  
+2. Check if sidecar injection is disabled in the Deployment:
```
Comment (tomekpapiernik) [inline docs/service-mesh/10-05-sidecar.md:20]:
```suggestion
2. Check if sidecar injection is disabled in the Pod's Deployment:
```

File: docs/service-mesh/10-05-sidecar.md [hunk 13]
```diff
@@ -4,4 +4,37 @@ type: Troubleshooting
...
+
+    ```bash
+    kubectl get deployments {DEPLOYMENT_NAME} -n {NAMESPACE} -o jsonpath='{ .spec.template.metadata.annotations }'
+    ```
+   
+   Sidecar injection is disabled if the output contains a line `sidecar.istio.io/inject:false`. Delete the label and restart the Pod.
```
Comment (tomekpapiernik) [inline docs/service-mesh/10-05-sidecar.md:26]:
```suggestion
   Sidecar injection is disabled if the output contains the `sidecar.istio.io/inject:false` line. Delete the label and restart the Pod to enable sidecar injection for the Deployment.
```

File: docs/service-mesh/10-05-sidecar.md [hunk 14]
```diff
@@ -4,4 +4,37 @@ type: Troubleshooting
...
+   
+3. Make sure Istio sidecar injector is running: 
+    
+    ```bash
+    kubectl describe pod -n istio-system -l app=sidecarInjectorWebhook
+    ```
+
+4. Make sure Istio sidecar injector can communicate with the Kubernetes API server. Search logs for any issues regarding connectivity:
```
Comment (tomekpapiernik) [inline docs/service-mesh/10-05-sidecar.md:34]:
```suggestion
4. Make sure Istio sidecar injector can communicate with the Kubernetes API server. Search logs for any connectivity issues. Run this command to get the logs:
```

File: docs/service-mesh/10-03-connection-refused.md (hunk 6 above)
Comment (colunira) [reply docs/service-mesh/10-03-connection-refused.md:9]: The direction is important in this case.

File: docs/kyma/10-04-common-errors.md [hunk 15]
```diff
@@ -61,4 +61,8 @@ kubectl apply -f https://raw.githubusercontent.com/kyma-project/kyma/{YOUR_KYMA_
 
 ## Maximum number of retries reached
 
-Kyma Installer retries installing a failed release a given number of times (default is 5). It stops the installation when it reaches the limit and returns this message: `Max number of retries reached during step {STEP_NAME}`. To start it again, check Kyma Installer logs to find out why this step failed, fix the issue, and follow the [update process](#installation-update-kyma).
+Kyma Installer retries installing a failed release a given number of times (default is 5). It stops the installation when it reaches the limit and returns this message: `Max number of retries reached during step {STEP_NAME}`. To start it again, check Kyma Installer logs to find out why this step failed, fix the issue, and run:
```
[the remark this answers is not in the document]
Comment (colunira) [reply docs/kyma/10-04-common-errors.md:64]: The message about reaching the maximum number of retries is available in the logs, so adding instructions on how to fetch the logs would be redundant.

File: docs/service-mesh/10-03-connection-refused.md [hunk 16]
```diff
@@ -1,11 +1,12 @@
 ---
-title: Connection refused errors after upgrade
+title: Connection refused errors
 type: Troubleshooting
 ---
 
-Mutual TLS (mTLS) is enabled in the Service Mesh by default. As a result, every element of the Service Mesh must have an Istio sidecar to allow TLS communication. If you don't want to use a sidecar, you can whitelist a service and disable TLS traffic for it by creating a [DestinationRule](https://istio.io/docs/reference/config/networking/destination-rule/).  
+Mutual TLS (mTLS) is enabled in the Service Mesh by default. As a result, every element of the Service Mesh must have an Istio sidecar with a valid TLS certificate to allow communication. Attempts to establish connection between a service with a sidecar and a service without a sidecar results in a `Connection reset by peer` or a `GOAWAY` response.
```
Comment (tomekpapiernik) [inline docs/service-mesh/10-03-connection-refused.md:6]:
```suggestion
Mutual TLS (mTLS) is enabled in the Service Mesh by default. As a result, every element of the Service Mesh must have an Istio sidecar with a valid TLS certificate to allow communication. Attempts to establish connection between a service with a sidecar and a service without a sidecar result in a `Connection reset by peer` or a `GOAWAY` response. 
```

Sorry, I missed that. Attempts -> result (plural)

File: docs/service-mesh/10-05-sidecar.md [hunk 17]
```diff
@@ -4,4 +4,37 @@ type: Troubleshooting
...
+  
+2. Check if sidecar injection is disabled in the Pod's Deployment:
+
+    ```bash
+    kubectl get deployments {DEPLOYMENT_NAME} -n {NAMESPACE} -o jsonpath='{ .spec.template.metadata.annotations }'
+    ```
+   
+   Sidecar injection is disabled if the output contains the `sidecar.istio.io/inject:false` line. Delete the label and restart the Pod to enable sidecar injection for the Deployment.
+   
+3. Make sure Istio sidecar injector is running:
```
Comment (tomekpapiernik) [inline docs/service-mesh/10-05-sidecar.md:28]:
```suggestion
3. Make sure that the Istio sidecar injector is running: 
```

File: docs/service-mesh/10-05-sidecar.md [hunk 18]
```diff
@@ -4,4 +4,37 @@ type: Troubleshooting
...
+   
+   Sidecar injection is disabled if the output contains the `sidecar.istio.io/inject:false` line. Delete the label and restart the Pod to enable sidecar injection for the Deployment.
+   
+3. Make sure Istio sidecar injector is running: 
+    
+    ```bash
+    kubectl describe pod -n istio-system -l app=sidecarInjectorWebhook
+    ```
+
+4. Make sure Istio sidecar injector can communicate with the Kubernetes API server. Search logs for any connectivity issues. Run this command to get the logs:
```
Comment (tomekpapiernik) [inline docs/service-mesh/10-05-sidecar.md:34]:
```suggestion
4. Make sure that the Istio sidecar injector can communicate with the Kubernetes API server. Search logs for any connectivity issues. Run this command to get the logs:
```
